lua/nvim-tree/actions/search-node.lua

```diff
@@ -0,0 +1,33 @@
+local utils = require'nvim-tree.utils'
+
+local M = {}
+
+function M.fn()
+  if not TreeExplorer then return end
+
+  local input_path = vim.fn.input("Search node: ", "", "file")
+  utils.clear_prompt()
+
+  local absolute_input_path = utils.path_join({
+    TreeExplorer.cwd,
+    input_path
+  })
+
+  local function search_node(nodes)
+    for _, node in ipairs(nodes) do
+      local matches = utils.str_find(absolute_input_path, node.absolute_path)
+      if matches and node.nodes then
+        if not node.open then
+          node.open = true
+          TreeExplorer:expand(node)
+        end
+        search_node(node.nodes)
+      end
+    end
+  end
+
+  search_node(TreeExplorer.nodes)
```

Comment:
you should redraw after that :)
also, might be interesting to copy more of the find-file algorithm and set the cursor on the found node ?